Mage.Sets/src/mage/cards/v/VesuvanDuplimancy.java

```diff
@@ -29,8 +31,14 @@ public final class VesuvanDuplimancy extends CardImpl {
     private static final FilterSpell filter
             = new FilterSpell("a spell that targets only a single artifact or creature you control");
 
+    private static final FilterPermanent subfilter = new FilterControlledPermanent("artifact or creature you control");
+
     static {
-        filter.add(VesuvanDuplimancyPredicate.instance);
+        subfilter.add(Predicates.or(
+                CardType.ARTIFACT.getPredicate(),
+                CardType.CREATURE.getPredicate()
+        ));
+        filter.add(new HasOnlySingleTargetPermanentPredicate(StaticFilters.FILTER_PERMANENT_ARTIFACT_OR_CREATURE));
```

Comment:
did you mean to use subfilter here?

Reply:
Wait no I don't think I made a mistake there.
filter is the "a spell that targets only a single artifact or creature you control", while subfilter is "artifact or creature you control"